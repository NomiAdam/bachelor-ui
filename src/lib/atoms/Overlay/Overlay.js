/* eslint-disable no-mixed-spaces-and-tabs */
import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import { IoIosArrowThinLeft } from 'react-icons/lib/io';
import { lightTheme, darkTheme } from '../../constants/theme';
import { FadeIn } from '../../transition/Fade/index';

const StyledWrapper = styled.div`
	display: ${ ({ isOpen }) => (isOpen ? 'flex' : 'none') }
	position: fixed;
	align-items: center;
	flex-direction: column;
	justify-content: ${ ({ verticalPosition }) => verticalPosition }
	top: 0;
	left: 0;
	width: 100%;
	height: 100%;
	z-index: 999;
	background-color: ${ ({ background }) => background };
`;

const InnerWrapper = styled.div``;

const StyledHeading = styled.h2`
	font-size: 1.5em;
	color: ${ darkTheme.black };
	margin: 0;
	padding: 60px 10px;
`;

const StyledCloseIcon = styled(IoIosArrowThinLeft)`
	position: absolute;
	cursor: pointer;
	top: 0;
	left: 0;
	margin: 0;
	font-size: 3rem;
	z-index: 999;
	color: ${ ({ backColor }) => backColor };
`;

const Overlay = ({
    isOpen, heading, handleClose, children, verticalPosition, background, displayHeading, backColor,
}) => (
    <StyledWrapper background={ background } isOpen={ isOpen } verticalPosition={ verticalPosition }>
        <InnerWrapper>
            {
                displayHeading && (
                    <StyledHeading>
                        {heading}
                    </StyledHeading>
                )
            }
            <FadeIn>
                {children}
            </FadeIn>
        </InnerWrapper>
        <StyledCloseIcon onClick={ handleClose } backColor={ backColor } />
    </StyledWrapper>
);

Overlay.propTypes = {
    /**
     * Whether selection window is open, if true window is open.
     */
    isOpen: PropTypes.bool,
    /**
     * String representation of heading
     */
    heading: PropTypes.string,
    /**
     * Function triggered upon Close button click, or on select.
     */
    handleClose: PropTypes.func,
    /**
     * String definition of vertical position of overlay
     */
    verticalPosition: PropTypes.string,
    /**
     * String definition of background color
     */
    background: PropTypes.string,
    /**
     * String definition of back Icon color
     */
    backColor: PropTypes.string,
    /**
     * Whether to display heading or not
     */
    displayHeading: PropTypes.bool,
};

Overlay.defaultProps = {
    isOpen: false,
    verticalPosition: 'center',
    background: lightTheme.white,
    displayHeading: true,
    backColor: darkTheme.black,
};

export { StyledWrapper };
export default Overlay;
