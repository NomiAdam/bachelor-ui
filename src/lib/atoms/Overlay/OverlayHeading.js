/* eslint-disable no-mixed-spaces-and-tabs */
import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import Heading from '../Heading/index';
import {IoIosArrowThinLeft} from 'react-icons/lib/io';
import { lightTheme, darkTheme } from '../../constants/theme';

const StyledWrapper = styled.div`
	display: flex;
	align-items: left;
	flex-direction: column;
	width: 100%;
	min-height: 150px;
	z-index: 999;
	border-bottom: 1px solid ${lightTheme.whiteBlue};
	background-color: ${lightTheme.lightGrey};
`;

const InnerWrapper = styled.div`
    padding: 0 20px;
    margin-top: 50px;
    margin-bottom: 25px;
`;

const StyledCloseIcon = styled(IoIosArrowThinLeft)`
	position: absolute;
	cursor: pointer;
	top: 0;
	left: 0;
	margin: 0;
	padding: 20px;
	font-size: 4rem;
	z-index: 10;
	color: ${darkTheme.black};
`;

const OverlayHeading = ({
                     heading, handleIconClick,
                 }) => (
    <StyledWrapper>
        <InnerWrapper>
           <Heading label={heading} component={'h2'}/>
        </InnerWrapper>
        <StyledCloseIcon onClick={handleIconClick}/>
    </StyledWrapper>
);

OverlayHeading.propTypes = {
    /**
     * String representation of heading
     */
    heading: PropTypes.string,
    /**
     * Function triggered upon Icon click.
     */
    handleIconClick: PropTypes.func,
};

OverlayHeading.defaultProps = {
    isOpen: false,
    verticalPosition: 'center',
};

export { StyledWrapper };
export default OverlayHeading;
