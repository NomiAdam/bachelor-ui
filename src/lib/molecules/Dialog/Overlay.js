import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import Container from '../../organisms/Container/index';

const StyledOverlayWrapper = styled(Container)`
  position: fixed;
  top: 0;
  left: 0;
  z-index: 99;
  background-color: rgba(0,0,0,.3);
`;

const refuseScroll = (e) => {
    e.preventDefault();
    return false;
};

const Overlay = ({ handleClick, children }) => (
    <StyledOverlayWrapper height="100vh" onClick={ handleClick } onScroll={ refuseScroll }>
        { children }
    </StyledOverlayWrapper>
);

Overlay.propTypes = {
    /**
     * onClick function handler
     */
    handleClick: PropTypes.func,
    /**
     * Children nodes
     */
    children: PropTypes.any,
};

export default Overlay;
