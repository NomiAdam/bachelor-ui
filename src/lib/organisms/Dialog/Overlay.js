import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import flexConstant from '../../constants/flex';
import Container from '../../molecules/Container/index';

const StyledOverlayWrapper = styled(Container)`
  position: fixed;
  top: 0;
  left: 0;
  z-index: 222;
  background-color: rgba(0,0,0,.3);
  display: ${ flexConstant.FLEX };
  justify-content: ${ flexConstant.POSITION_CENTER };
  align-items: ${ flexConstant.POSITION_CENTER };  
  overflow-y: auto;
`;

const refuseScroll = (e) => {
    e.preventDefault();
    return false;
};

const Overlay = ({ handleClick, children, secondary }) => (
    <StyledOverlayWrapper secondary={ secondary } height="100vh" onClick={ handleClick } onScroll={ refuseScroll }>
        { children }
    </StyledOverlayWrapper>
);

Overlay.propTypes = {
    /**
     * onClick function handler
     */
    handleClick: PropTypes.func,
    /**
     * Children nodes
     */
    children: PropTypes.any,
};

export default Overlay;
