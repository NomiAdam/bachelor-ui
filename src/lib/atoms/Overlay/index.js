import Overlay from './Overlay';
import OverlayHeading from './OverlayHeading';

export {
    OverlayHeading,
};

export default Overlay;
