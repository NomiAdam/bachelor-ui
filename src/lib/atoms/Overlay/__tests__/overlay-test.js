/* eslint-disable no-undef */
import React from 'react';
import Overlay, {StyledWrapper} from '../Overlay';
import { shallow } from 'enzyme';
import 'jest-styled-components';

describe('Overlay', () => {
    it('should be defined', () => {
        expect(Overlay).toBeDefined();
    });
    it('should match the snapshot', () => {
        expect(shallow(<Overlay />)).toMatchSnapshot();
    });
    it('should be of type StyledOverlayWrapper', () => {
        expect(shallow(<Overlay/>).type()).toBe(StyledWrapper);
    });
    it('should have display `none` when isOpen is set to false', () => {
        const wrapper = shallow(<Overlay />);
        expect(wrapper.find(StyledWrapper)).toHaveStyleRule('display', 'none');
    });
    it('should have display `flex` when isOpen is set to false', () => {
        const wrapper = shallow(<Overlay isOpen />);
        expect(wrapper.find(StyledWrapper)).toHaveStyleRule('display', 'flex');
    });
});
