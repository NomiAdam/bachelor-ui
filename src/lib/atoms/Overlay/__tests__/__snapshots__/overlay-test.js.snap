// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Overlay should match the snapshot 1`] = `
<styled.div
  background="#ffffff"
  isOpen={false}
  verticalPosition="center"
>
  <styled.div>
    <styled.h2 />
    <Styled(styled.div) />
  </styled.div>
  <Styled(IoIosArrowThinLeft)
    backColor="#000000"
  />
</styled.div>
`;
